### ISSUE TITLE
Issue 66056: openh264:decoder_fuzzer: Integer-overflow in WelsDec::ParseSliceHeaderSyntaxs

### ISSUE THREAD

**Issue by BenzhengZhang** [cisco/openh264#3745]
https://bugs.chromium.org/p/oss-fuzz/issues/detail?id=66056&q=label%3AProj-openh264&can=2

**Comment by BenzhengZhang** [comment]
@tyan0 Recently fuzzy test reported two decoder bugs, another #3746 Could you pls take a look this issue? thank you.

**Comment by tyan0** [comment]
Could you please let me know how to reproduce the issues?

**Comment by BenzhengZhang** [comment]
@tyan0 you can refer to https://google.github.io/oss-fuzz/advanced-topics/reproducing/

**Comment by BenzhengZhang** [comment]
> Could you please let me know how to reproduce the issues?
I remembered, you may not be able to open the bug report link. https://bugs.chromium.org/p/oss-fuzz/issues/detail?id=66056&q=label%3AProj-openh264&can=2 

Reproducer Testcase:  
[clusterfuzz-testcase-minimized-decoder_fuzzer-4680815741042688 (1).zip](https://github.com/cisco/openh264/files/15105226/clusterfuzz-testcase-minimized-decoder_fuzzer-4680815741042688.1.zip)

**Comment by tyan0** [comment]
I'm very sorry for extremely long delay.

Finally, I could reproduce the problems with the testcases:
clusterfuzz-testcase-minimized-decoder_fuzzer-4680815741042688 (#3745)
clusterfuzz-testcase-minimized-decoder_fuzzer-6315387293597696 (#3746)

I'll look into these problems.

**Comment by BenzhengZhang** [comment]
So this is a false positive, right? @tyan0

**Comment by tyan0** [comment]
No, it is not. Current code works as expected on the most system (where two's complement is used for negative value) with gcc. However, C++ standard inhibit expecting the certain result on overflow on signed integer (the result is **UNDEFINED**).

Therefore, in some systems (https://stackoverflow.com/questions/12276957/are-there-any-non-twos-complement-implementations-of-c), the code does not work as expected. So, the current code is not portable. It depends on system.

I am not sure that openh264 should be portable even for such systems.

**Comment by fermay** [comment]
> As for #3745, the code in decoder_core.cpp expects that the arithmetic overflow in int32_t results into wrapped-around result. Oss-fuzz complains about that. C/C++ standard states the overflowed result is undefined for signed integer.

i also think not need to modify this code,   2 reasons: 1.  if modify it, it will cause the code more hard understand. 2.  msb and lsb are hard to exceed int32_t  in actually.
